reader/xml/decoder.go

```diff
@@ -0,0 +1,47 @@
+package xml // import "miniflux.app/reader/xml"
+
+import (
+	"bytes"
+	"encoding/xml"
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	"miniflux.app/reader/encoding"
+)
+
+// GetDecoder returns a XML decoder that filters out illegal characters
+func GetDecoder(data io.Reader) *xml.Decoder {
```

Comment:
It would be better to call this function `NewDecoder` like the original one. Seems more idiomatic.